### PR TITLE
Release 3.0 beta 1

### PR DESCRIPTION
**Pull request by CharliePoole**
This is ready for review. The plan is to merge it and then do the release, so it should be reviewed for readiness to release as well as correctness.

The fix to the msi install (last commit) is intended to be temporary. Having to edit the projects to put in "3.0.0-beta-1" is certainly not a good long-term solution but I wanted to get the release out. The use of DisplayVersion as a pre-processor variable is less problematic since it appears in a VS dialog, but we will want to review that as well. Basically, we have to first decide whether to use an approach where we input the package version in some way as we did in the NUnit 2.x scripts or pull it out of one of the assemblies in each package. That said, I think we're OK for the moment and I didn't want to do anything major to the script at the last moment.

I'll upload the binaries somewhere in the next hour and post URL here.

### CODE REVIEW

File: CHANGES.txt [hunk 1]
```diff
@@ -1,5 +1,70 @@
 ﻿NOTE: A combined description of changes since NUnit 2.6.3 is under development along with a summary statement of major features. Both will be available in the first Beta release.
 
+NUnit 3.0.0 Beta 1 - In Process
```
Comment (rprouse) [inline CHANGES.txt]: I assume `In Process` will be changed to the release date or should that have been done?

File: CHANGES.txt [hunk 2]
```diff
@@ -1,5 +1,70 @@
...
+ * There is now a master windows installer for the framework, engine and console runner.
+
+Framework
+
+ * We no longer create a separate framework build for .NET 3.5. The 2.0 and
+   3.5 builds were essentially the same, so the former should now be used 
+   under both runtimes.
+ * A new Constraint, DictionaryContainsKeyConstraint, may be used to test
+   that a specified key is present in a dictionary.
+ * LevelOfParallelizationAttribute has been renamed to LevelOfParallelism.
```
Comment (rprouse) [inline CHANGES.txt]: Attribute should be removed from the first or added to the second for consistency.

File: CHANGES.txt [hunk 3]
```diff
@@ -1,5 +1,70 @@
...
+
+Engine
+
+ * The engine API has now been finalized. It permits specifying a minimum 
+   version of the engine that a runner is able to use. The best installed 
+   version of the engine will be loaded. Third-party runners may override
+   the selection process by including a copy of the engine in their 
+   installation directory and specifying that it must be used.
+ * The V2 framework driver now uses the event listener and test listener
+   passed to it by the runner. This which corrects several outstanding issues
```
Comment (rprouse) [inline CHANGES.txt]: Remove `which`

File: CHANGES.txt [hunk 4]
```diff
@@ -1,5 +1,70 @@
...
+   caused by events not being received and allows selecting V2 tests to
+   be run from the command-line, in the same way that V3 tests are selected.
+
+Console
+
+ * The console now defaults to not using shadowcopy and the --noshadow option
```
Comment (rprouse) [inline CHANGES.txt]: There was no --noshadow option in NUnit 3

File: CHANGES.txt (hunk 1 above)
Comment (CharliePoole) [inline CHANGES.txt]: Yes, will be. I wasn't sure how long reviewing might take. I'd like you to have an opportunity to actually run the code.

File: install/framework/framework.wxs [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
 	<Product Id="*" 
-             Name="NUnit Framework !(bind.FileVersion.nunit.framework.45.dll)" 
+             Name="NUnit Framework $(var.DisplayVersion)"
```
Comment (rprouse) [inline install/framework/framework.wxs]: Should the various uses of `DisplayVersion` include the beta modifier?

File: CHANGES.txt (hunk 2 above)
Comment (CharliePoole) [inline CHANGES.txt]: Added.

**Comment by rprouse** [comment]
All minor points. Feel free to change as you see fit and commit/merge when ready.

File: CHANGES.txt (hunk 4 above)
Comment (CharliePoole) [inline CHANGES.txt]: Good catch. Belongs in the document that compares 3.0 to 2.6.4.

File: install/framework/framework.wxs (hunk 5 above)
Comment (CharliePoole) [inline install/framework/framework.wxs]: In the past, the dialogs for our installs didn't say "alpha" or "beta" so I kept that consistent.

Comment (rprouse) [inline install/framework/framework.wxs]: If that is the case, then how will users know if the have the beta or the final release installed by looking in Add/Remove Programs? Look at the version column?

Comment (CharliePoole) [inline install/framework/framework.wxs]: Hmmm... I never worried about that before. I suppose I assume that installers of alphas and betas are quite on top of things and know what's on their systems. Certainly, there has never been any comment about it in the past.

As an experiment, I installed the ancient NUnit 2.4 beta 2 and the new NUnit.Framework 3.0 beta 1. For 2.4, I could see the version number (2.3.2693) and could figure out that it's beta 2 by looking back at the release notes. For 3.0, I see version number 3.0.5560.36863 so I could figure that out if we listed it somewhere.

Your point is well taken. Maybe we should make this a bit easier.

**Pull request merged**